### PR TITLE
LibJS: Accept symbol property in ObjectPrototype::hasOwnProperty

### PR DESCRIPTION
**Pull request by IdanHo**
This is used by discord.com and allowed by the specification (https://tc39.es/ecma262/#sec-topropertykey)

### CODE REVIEW

**Comment by bgianfo** [comment]
Does it make sense to add a test to make sure this stays working / spec conforming?

**Comment by IdanHo** [comment]
> Does it make sense to add a test to make sure this stays working / spec conforming?

Yep! done. (While i was there i also found a small inconsistency with the specification in JSON.stringify, so added a commit for that as well :)

File: Userland/Libraries/LibJS/Tests/builtins/Object/Object.prototype.hasOwnProperty.js [hunk 1]
```diff
@@ -10,4 +10,9 @@ test("basic functionality", () => {
     o.undefined = 2;
     expect(o.hasOwnProperty()).toBeTrue();
     expect(o.hasOwnProperty(undefined)).toBeTrue();
+
+    var test_symbol = Symbol("real");
+    o[test_symbol] = 3;
+    expect(o.hasOwnProperty(test_symbol)).toBeTrue();
+    expect(o.hasOwnProperty(Symbol("fake"))).toBeFalse();
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/Object/Object.prototype.hasOwnProperty.js:17]: Please use pascalCase in JavaScript code, snake_case doesn't mix nicely with the built-in function names.

```suggestion
    var testSymbol = Symbol("real");
    o[testSymbol] = 3;
    expect(o.hasOwnProperty(testSymbol)).toBeTrue();
    expect(o.hasOwnProperty(Symbol("fake"))).toBeFalse();
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibJS/Runtime/JSONObject.cpp [hunk 2]
```diff
@@ -215,6 +215,8 @@ String JSONObject::serialize_json_object(GlobalObject& global_object, StringifyS
     Vector<String> property_strings;
 
     auto process_property = [&](const PropertyName& key) {
+        if (key.is_symbol())
+            return;
```
[the remark this answers is not in the document]
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/JSONObject.cpp:219]: for sure

File: Userland/Libraries/LibJS/Tests/builtins/Object/Object.prototype.hasOwnProperty.js (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Tests/builtins/Object/Object.prototype.hasOwnProperty.js]: fixed! ill try to remember that for next time :)

**Pull request merged**